share/nid.go

```diff
@@ -0,0 +1,26 @@
+package share
+
+import (
+	"fmt"
+
+	appns "github.com/celestiaorg/celestia-app/pkg/namespace"
+	"github.com/celestiaorg/nmt/namespace"
+)
+
+// NewNamespaceV0 takes variable size byte slice anc creates version 0 Namespace ID.
+// The bytes slice must be <= 10 bytes.
+func NewNamespaceV0(subNId []byte) (namespace.ID, error) {
```

Comment:
@rootulp, we introduced this helper so that users don't have to struggle with defining the exact correct namespace size nor importing the app to get the size. It would be nice if the namespace in app had this user-friendly helper

Reply:
> with defining the exact correct namespace size

the namespace that gets used will still have a fixed length so even if NewNamespaceV0 is called with a subNId length of less than 10, the caller must know that the usable portion of the namespace ID is 10 bytes.

See https://go.dev/play/p/jNv2FkEG-ew

celestia-app uses a fixed namespace in tests like [here](https://github.com/celestiaorg/celestia-app/blob/ddaa261b9b0dec5f5ea10dbfab4c8a03c95d7e72/app/test/check_tx_test.go#L23) which I think may be more readable / understandable than the approach with `subNId`.

Reply:
> the caller must know that the usable portion of the namespace ID is 10 bytes.

That's a documentation problem(the func is not perfectly documented, like NMT). 

What I'm afraid I have to disagree with is that we should force the user to give _the exact size_ of the namespace ID. DevX-wise, this is horrible.
 
Let's count how many things a user with zero context should do to create a namespace:
 * import consts
 * find(there is insane number of consts now) and understand which const is correct 
 * preallocate the slice with the constant
 * fill in his wanted NamespaceID
 * import bytes
 * fill in the slice with zeroes.
 
In this PR's approach, you should know where to get the constructor(documentation issue) and give the wanted NamespaceID. That's all. Implementation details of how many bytes there are actually are not necessary for simple users

Reply:
celestia-app already contains https://github.com/celestiaorg/celestia-app/blob/ddaa261b9b0dec5f5ea10dbfab4c8a03c95d7e72/pkg/namespace/namespace.go#L41-L43 so celestia-node can do:

```diff
		if !sameNID {
+			nid := appns.MustNewV0(tmrand.Bytes(appns.NamespaceVersionZeroIDSize)).Bytes()
-			nid, err := share.NewNamespaceV0(tmrand.Bytes(7))
-			if err != nil {
-				return nil, err
-			}
			appBlob.NamespaceVersion = nid[0]
			appBlob.NamespaceID = nid[1:]
		}
```

however, I see your point that this approach relies on consumers importing a const from celestia-app. I'm open to adding the following function to celestia-app's namespace package given it seems slightly easier to use from DevX perspective:

```go
// NewV0 returns a new namespace with version 0 and id based on subID. subID
// must be <= 10 bytes.
//
// Note: the namespace ID will be right-padded with 0s if subID is < 10 bytes.
func NewV0(subID []byte) (Namespace, error) {
	if lenSubID := len(subID); lenSubID > NamespaceVersionZeroIDSize {
		return Namespace{}, fmt.Errorf("subID must be <= %v, but it was %v bytes", NamespaceVersionZeroIDSize, lenSubID)
	}

	id := make([]byte, NamespaceIDSize)
	copy(id[NamespaceVersionZeroPrefixSize:], subID)

	ns, err := New(NamespaceVersionZero, id)
	if err != nil {
		return Namespace{}, err
	}

	return ns, nil
}
```

Note: it has been slightly modified from the `NewNamespaceV0` in this PR because it returns the `Namespace` type rather than `namespace.ID`. Is that okay for your use case? The motivation for the change is that this function will be defined in the celestia-app `namespace` package which ideally has functions that deal with [Namespace](https://github.com/celestiaorg/celestia-app/blob/ddaa261b9b0dec5f5ea10dbfab4c8a03c95d7e72/pkg/namespace/namespace.go#L8-L11)